### PR TITLE
import_error_alert_emails

### LINKED ISSUE #10968: import error alert emails
**Description**

Send alert emails on import errors.

**Use case / motivation**
This is a feature will be especially useful for admins who use `ClusterPolicyViolation` or those who have pesky folks that push bad DAGs cluttering the UI with notifications that should be triaged to the on-call.

**Related Issues**
I believe that the `import_errors` is a sad place in the airflow codebase and now might be the time to refactor this.
I have questions about how to best keep track of import error emails we have sent (as to not spam the recipients with emails every scheduler loop).

It seems to me we can either keep history of import errors in the existing table (this would require a fundamental change to the scheduler logic which clears this table on each loop and a change to how the UI uses this table) or create a new table (e.g. `import_error_history` which has a bool field for email_sent) for this purpose.

I'm curious to hear from community if there are other use cases / improvements we should consider in refactoring the import errors.

cc: @mik-laj

### PR DESCRIPTION
**Pull request by jharner11235**
This adds functionality to notify a list of emails if a dag fails to import, and uses a variable in airflow.cfg to assign emails to be notified (instead of attempting to parse a broken dag for emails, which may at times be unviable)

closes: #10968
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by jharner11235** [comment]
I'm looking for some guidance in how to apply tests, I've yet to do any testing whatsoever near the scale present here and am quite lost.

And of course, anything goes on the already-written code as well :) happy to make any changes seen fit

**Comment by ephraimbuddy** [comment]
I don't think it's cool to get alerts of broken dags every 5mins? WDYT @jharner11235

**Comment by jharner11235** [comment]
> I don't think it's cool to get alerts of broken dags every 5mins? WDYT @jharner11235

I didn't get an alert every 5 minutes in any of my testing, given where the changes are made if alerts were being sent every 5mins then alerts would also be inserted into the database every 5mins, and I didn't see that occurring either

File: airflow/config_templates/config.yml [hunk 1]
```diff
@@ -1477,6 +1477,20 @@
       type: boolean
       example: ~
       default: "True"
+    - name: email_on_dag_import_failure
+      description: |
+        Whether email alerts should be sent when a dag fails to import
+      version_added: ~
+      type: boolean
+      example: ~
+      default: "True"
```
Comment (ashb) [inline airflow/config_templates/config.yml]: We don't need a setting and a list of emails -- empty = don't email.

File: airflow/dag_processing/processor.py [hunk 2]
```diff
@@ -549,6 +549,17 @@ def update_import_errors(session: Session, dagbag: DagBag) -> None:
                     synchronize_session='fetch',
                 )
             else:
+                emails: Set[str] = set(conf.get(section='email', key='dag_import_failure_email_to').split())
+                if emails and conf.getboolean(
+                    section='email', key='email_on_dag_import_failure', fallback=True
+                ):
```
Comment (ashb) [inline airflow/dag_processing/processor.py]:
```suggestion
                emails: Set[str] = set(conf.get(section='email', key='dag_import_failure_email_to', fallback='').split())
                if emails:
```

File: airflow/dag_processing/processor.py [hunk 3]
```diff
@@ -549,6 +549,17 @@ def update_import_errors(session: Session, dagbag: DagBag) -> None:
...
+                    try:
+                        email_content = f"""File {filename} has failed to import due to:\n
+                        <pre><code>{stacktrace}<code></pre>
```
Comment (ashb) [inline airflow/dag_processing/processor.py:574]: I'm not sure if the emails are plain text or html, though I assume plain text. So this doesn't seem right.

**Review by ashb — CHANGES_REQUESTED** [review]

File: airflow/dag_processing/processor.py (hunk 3 above)
Comment (jharner11235) [reply airflow/dag_processing/processor.py:574]: The emails show up and look as expected, it does seem to be HTML

File: airflow/config_templates/config.yml (hunk 1 above)
Comment (jharner11235) [reply airflow/config_templates/config.yml]: oh yeah, that's fair. I'll make the adjustment

File: airflow/dag_processing/processor.py (hunk 2 above)
Comment (jharner11235) [reply airflow/dag_processing/processor.py]: suggestion taken but wrapped in commit with other changes

**Pull request closed without merging**